### PR TITLE
Implement permission() for custom SQL processors

### LINKED ISSUE #19087: Ad-hoc processors created by SQL module does not use SecuredFunctions
One can obtain a non-secured ServiceFactory using those processors

```
Class<?> aClass = Class.forName("com.hazelcast.jet.sql.impl.connector.map.UpdateProcessorSupplier");
        Constructor<?>[] declaredConstructors = aClass.getDeclaredConstructors();
        declaredConstructors[1].setAccessible(true);
        ProcessorSupplier instance = (ProcessorSupplier)declaredConstructors[1].newInstance(name, null);
        AbstractTransformUsingServiceP processor = ((List<AbstractTransformUsingServiceP>)instance.get(1)).get(0);
        Field serviceContextField = AbstractTransformUsingServiceP.class.getDeclaredField("serviceFactory");
        serviceContextField.setAccessible(true);
        ServiceFactory serviceFactory = (ServiceFactory)serviceContextField.get(processor);
```

### PR DESCRIPTION
**Pull request by gierlachg**
Fixes #19087

### CODE REVIEW

**Comment by viliam-durina** [comment]
I don't think we need this. The security is handled here: https://github.com/hazelcast/hazelcast/blob/master/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/JetSqlBackend.java#L307-L336
I think it's even more robust - harder to forget about some case

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java [hunk 1]
```diff
@@ -94,6 +100,11 @@ public void init(@Nonnull Context context) {
                 .thenApply(m -> Traversers.empty());
     }
 
+    @Override
+    public Permission permission() {
+        return new MapPermission(mapName, ACTION_CREATE, ACTION_PUT, ACTION_REMOVE);
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java:105]: Can it really remove? https://github.com/hazelcast/hazelcast/blob/b1a5fd7fdaa6d29b76d08577cc57a7e43e034b68/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdatingEntryProcessor.java#L76

**Review by viliam-durina — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java (hunk 1 above)
Comment (gierlachg) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java:105]: From what I know UPDATE=REMOVE + PUT @gurbuzali

Comment (viliam-durina) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/UpdateProcessorSupplier.java:105]: We can't update the key, so it's just PUT i guess. And we specifically don't allow updating the value to `null`, which would be REMOVE. So I think it's just PUT.

**Review by gurbuzali — APPROVED** [review]

**Comment by gierlachg** [comment]
run-lab-run

**Pull request merged**